### PR TITLE
Fix problem with parse pair element ("key") and make mapOverlays function public

### LINKED ISSUE #216: StyleUrl for GMUPlacemark not matching up with any GMUStyle within the _styles property of GMUGeometryRenderer.
I have found that when the below kml document is parsed by GMUKMLParser, the resulting GMUPlacemark object does not have a valid string within the StyleUrl property that references a GMUStyle object.  Thus no style is applied to the given placemark despite being set and exported via Google Earth Pro desktop Mac app. 

```
<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2" xmlns:gx="http://www.google.com/kml/ext/2.2" xmlns:kml="http://www.opengis.net/kml/2.2" xmlns:atom="http://www.w3.org/2005/Atom">
<Document>
	<name>KW Path.kml</name>
	<Style id="sn_ylw-pushpin">
		<IconStyle>
			<scale>1.1</scale>
			<Icon>
				<href>http://maps.google.com/mapfiles/kml/pushpin/ylw-pushpin.png</href>
			</Icon>
			<hotSpot x="20" y="2" xunits="pixels" yunits="pixels"/>
		</IconStyle>
		<LineStyle>
			<color>ff00c2ff</color>
			<width>2</width>
		</LineStyle>
	</Style>
	<Style id="sh_ylw-pushpin">
		<IconStyle>
			<scale>1.3</scale>
			<Icon>
				<href>http://maps.google.com/mapfiles/kml/pushpin/ylw-pushpin.png</href>
			</Icon>
			<hotSpot x="20" y="2" xunits="pixels" yunits="pixels"/>
		</IconStyle>
		<LineStyle>
			<color>ff00c2ff</color>
			<width>2</width>
		</LineStyle>
	</Style>
	<StyleMap id="msn_ylw-pushpin">
		<Pair>
			<key>normal</key>
			<styleUrl>#sn_ylw-pushpin</styleUrl>
		</Pair>
		<Pair>
			<key>highlight</key>
			<styleUrl>#sh_ylw-pushpin</styleUrl>
		</Pair>
	</StyleMap>
	<Placemark>
		<name>KW Path</name>
		<styleUrl>#msn_ylw-pushpin</styleUrl>
		<LineString>
			<tessellate>1</tessellate>
			<coordinates>
				-95.40955451587095,29.68319223122121,0 -95.41162049810978,29.68311038759943,0 
			</coordinates>
		</LineString>
	</Placemark>
</Document>
</kml>
```

### PR DESCRIPTION
**Pull request by ArtemBoboshko**
Fixes #216. I noticed that GMUKMLParser don't parse and don't save GMUPair objects to GMUStyleMap.
So i've added this logic to the GMUKMLParser.

Also, I faced with an issue that I can't make my overlays tappable after I used GMUGeometryRenderer to render geometries.
I found `mapOverlays` function that are not used. I think this function were made to be public but it's not in h file for some reasons.
I've made it public.

@domesticmouse Looking forward to have it merged.
Thanks.

 Test passing
  CLA submitted

### CODE REVIEW

**Comment by ArtemBoboshko** [comment]
Fixes #216. I noticed that GMUKMLParser don't parse and don't save GMUPair objects to GMUStyleMap.
So i've added this logic to the GMUKMLParser.

Thanks.

**Comment by ArtemBoboshko** [comment]
@domesticmouse Looking forward to having it checked.

**Comment by arriolac** [comment]
@googlebot

File: src/Geometry/GMUGeometryRenderer.h [hunk 1]
```diff
@@ -30,6 +30,10 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @interface GMUGeometryRenderer : NSObject
 
+/**
+ * The overlays array returned from the GMUGeometryRenderer, use after render function.
+ */
+- (NSArray<GMSOverlay *> *)mapOverlays;
```
Comment (arriolac) [inline src/Geometry/GMUGeometryRenderer.h]: Please revert this change. The `mapOverlays` function defined in `GMUGeometryRenderer.m` was not designed to be exposed in the public interface. 

If you need access to this, I recommend creating a class extension similar to [GMUGeometryRenderer+Testing.h](https://github.com/googlemaps/google-maps-ios-utils/blob/master/src/Geometry/GMUGeometryRenderer+Testing.h) to expose this method.

File: CONTRIBUTORS.txt [hunk 2]
```diff
@@ -15,3 +15,4 @@ Michael Lapuebla <chainedtothewoods@gmail.com>
 Daniel Kostrzynski <kostrzynski@google.com>
 Christian Ihle <blurpy@gmail.com>
 Gareth Pearce <garethpearce@google.com>
+Artem Boboshko <boboshkoa@meta.ua>
```
Comment (arriolac) [inline CONTRIBUTORS.txt]: Looks like there are some conflicts in this file that need to be addressed.

**Comment by arriolac** [comment]
Opened #259 instead.
